### PR TITLE
Set env variables usage instead of getopts

### PR DESCRIPTION
**Pull request by niconosenzo**
Efforts part of Issue #504
Refactoring stage: Argument variable values as env variables instead of getops strategy
Script: **build-controller-image.sh**

Description of changes:
Removed getops strategy in favor of a fixed number of arguments (aws service name in this case) and env variables for the previous values passed through arguments

### CODE REVIEW

File: scripts/build-controller-image.sh [hunk 1]
```diff
@@ -17,43 +16,27 @@ check_is_installed docker
...
-      ;;
-    \? )
-        echo "${USAGE}" 1>&2
-        exit
-      ;;
-  esac
-done
+Environment variables:
+  QUIET:                    Build controller container image quietly (<true|false> Default: false)
+  AWS_SERVICE_DOCKER_IMG:   Controller container image tag (Default: ack-<service>-controller:$VERSION)
```
Comment (jaypipes) [inline scripts/build-controller-image.sh:29]: Let's go ahead and place the `Default: XXXX` part on a separate line, similar to how these are done:

https://github.com/aws/aws-controllers-k8s/blob/1ca9215edc0b5883075712e0dbeb7fd5374a8bd0/scripts/helm-package-controller.sh#L24-L27

**Review by jaypipes — CHANGES_REQUESTED** [review]
teeny little request, see inline :)

File: scripts/build-controller-image.sh [hunk 2]
```diff
@@ -17,43 +16,27 @@ check_is_installed docker
 
 USAGE="
 Usage:
```
Comment (A-Hilaly) [inline scripts/build-controller-image.sh:18]: One more thing we can take care of in this PR, is changing how this script is called in other files. PTAL at [./kind-build-test.sh](https://github.com/aws/aws-controllers-k8s/blob/main/scripts/kind-build-test.sh#L140-L145) and [./publish-controller-image.sh](https://github.com/aws/aws-controllers-k8s/blob/main/scripts/publish-controller-image.sh#L57-L60)

**Review by A-Hilaly — COMMENTED** [review]
LGTM overall 👍 . I left few comments :)

File: scripts/build-controller-image.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (niconosenzo) [reply scripts/build-controller-image.sh:29]: Understood, then changing DEFAULT_AWS_SERVICE_DOCKER_IMG to "$AWS_SERVICE-$VERSION" as well as the Usage string. Will take care of the default var assignation in a separate PR.

File: scripts/build-controller-image.sh (hunk 2 above)
Comment (niconosenzo) [reply scripts/build-controller-image.sh:18]: @A-Hilaly  public-controller-image.sh calls this script setting the svc docker image tag to AWS_SERVICE_DOCKER_IMG_TAG, this script expects var name AWS_SERVICE_DOCKER_IMG instead, there seems to be some confusion and the script (public-controller-image.sh) uses both variable names indistinctly, is that correct ?

Comment (A-Hilaly) [reply scripts/build-controller-image.sh:18]: Nice catch. This is correct - `AWS_SERVICE_DOCKER_IMG` is the variable that should be used to call `./build-controller-image.sh`. the other variable `AWS_SERVICE_DOCKER_IMG_TAG` should only be used to construct the variable `AWS_SERVICE_DOCKER_IMG`

Comment (A-Hilaly) [reply scripts/build-controller-image.sh:18]: Something like this makes more sense
```bash
: "${AWS_SERVICE_DOCKER_IMG:="$DOCKER_REPOSITORY:$DEFAULT_AWS_SERVICE_DOCKER_IMG_TAG"}"

$SCRIPTS_DIR/build-controller-image.sh -s "$AWS_SERVICE" -i "$AWS_SERVICE_DOCKER_IMG"
```

**Review by A-Hilaly — APPROVED** [review]
/lgtm thanks @niconosenzo!

**Review by jaypipes — APPROVED** [review]
:+1: thanks @niconosenzo! Welcome to the ACK contributor club :)

**Pull request merged**